src/components/componentDiff/ComponentDiffTab.js

```diff
@@ -219,6 +221,79 @@ const ComponentVersionDiff = ({
     rightVersion: rightVersion,
   })
 
+  const determineMainSourceForComponent = (component) => {
+    if(!component.sources || component.sources.length === 0) {
+      return '<unknown>'
+    }
+    const mainSource = component.sources.find(source => {
+      const label = source.labels?.find(l => l.name === knownLabelNames.mainSource)
+      return label && label.value && label.value['repository-classification'] === 'main'
+    })
+    if (mainSource) {
+      return mainSource.access?.repoUrl || '<unknown>'
+    }
+    return component.sources[0].access?.repoUrl || '<unknown>'
+
+  }
+
+  const formatResourceName = (resource, component) => {
```

Comment:
`format` is not really meaningful, how about something like `prefixKnownResourceTypes` ?